src/commands/deployWorkspaceProject/deploymentConfiguration/workspace/AzureResourceVerifyStepBase.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { GenericTreeItem, activityFailContext, activityFailIcon, activitySuccessContext, activitySuccessIcon } from "@microsoft/vscode-azext-utils";
+import { type Progress } from "vscode";
+import { ExecuteActivityOutputStepBase, type ExecuteActivityOutput } from "../../../../utils/activity/ExecuteActivityOutputStepBase";
+import { createActivityChildContext } from "../../../../utils/activity/activityUtils";
+import { localize } from "../../../../utils/localize";
+import { type DeploymentConfigurationSettings } from "../../settings/DeployWorkspaceProjectSettingsV2";
+import { type WorkspaceDeploymentConfigurationContext } from "./WorkspaceDeploymentConfigurationContext";
+
+export abstract class AzureResourceVerifyStepBase extends ExecuteActivityOutputStepBase<WorkspaceDeploymentConfigurationContext> {
+    public abstract priority: number;
+
+    protected abstract resourceType: 'resource group' | 'container app' | 'container registry';
+    protected abstract deploymentSettingsKey: string;
+    protected abstract contextKey: string;
+
+    protected async executeCore(context: WorkspaceDeploymentConfigurationContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        this.options.shouldSwallowError = true;
+        progress.report({ message: localize(`verifyingResourceType`, 'Verifying {0}...', this.resourceType) });
+
+        const settings: DeploymentConfigurationSettings | undefined = context.deploymentConfigurationSettings;
+        if (!settings?.[this.deploymentSettingsKey]) {
+            return;
+        }
+
+        await this.verifyResource(context);
+
+        if (!context?.[this.contextKey]) {
+            // We shouldn't ever see this error directly
```

Comment:
Maybe leave a comment how this gets set when verifying (thus why it shouldn't ever be seen)